pkg/client/connection.go

```diff
@@ -96,4 +96,10 @@ type Connection interface {
 
 	// Unsubscribe unsubscribes from a destination
 	Unsubscribe(destination string) error
+
+	// Requestor API
+	NewRequestor(spec RequestorSpec) Requestor
```

Comment:
I think this should return the requestor object and an error, as there may be errors detected at this point, like a missing field in the spec, etc.